--- a/qtwebengine/src/3rdparty/chromium/base/base.gypi	2015-06-29 22:09:56.000000000 +0200
+++ b/qtwebengine/src/3rdparty/chromium/base/base.gypi	2015-07-13 16:33:49.412000000 +0200
@@ -744,14 +744,6 @@
           4018,
         ],
         'target_conditions': [
-          ['(<(desktop_linux) == 0 and <(chromeos) == 0) or >(nacl_untrusted_build)==1', {
-              'sources/': [
-                ['exclude', '^nix/'],
-              ],
-              'sources!': [
-                'atomicops_internals_x86_gcc.cc',
-              ],
-          }],
           # We need to include xdg for embedded linux,
           # but we indeed do not want atomicops for x86.
           ['qt_os == "embedded_linux"', {
