--- a/qtwebengine/src/3rdparty/chromium/base/base.gypi	2015-09-04 14:40:57.000000000 +0200
+++ b/qtwebengine/src/3rdparty/chromium/base/base.gypi.new	2015-09-24 16:35:45.863120999 +0200
@@ -773,11 +773,6 @@
               'memory/shared_memory_posix.cc',
             ],
           }],
-          ['(<(desktop_linux) == 0 and <(chromeos) == 0) or >(nacl_untrusted_build)==1', {
-              'sources/': [
-                ['exclude', '^nix/'],
-              ],
-          }],
           # We need to include xdg for embedded linux,
           # but we indeed do not want atomicops for x86.
           ['qt_os == "embedded_linux"', {
