--- plex/CMakeLists.txt	2013-09-05 14:41:36.000000000 +0200
+++ plex/CMakeLists.txt.new	2013-10-25 17:51:05.669777568 +0200
@@ -115,9 +115,15 @@
     set(WORKDIR ${CMAKE_CURRENT_BINARY_DIR})
   endif(WIN32)
 
+  if(TEXTUREPACKERPATH)
+    set(TEXTUREPACKER_EXE ${TEXTUREPACKERPATH})
+  else(TEXTUREPACKERPATH)
+    set(TEXTUREPACKER_EXE $<TARGET_FILE:TexturePacker>)
+  endif(TEXTUREPACKERPATH)
+
   add_custom_command(
     OUTPUT Textures_MediaStream.xbt
-    COMMAND $<TARGET_FILE:TexturePacker> -input ${root}/addons/skin.mediastream/Media -output ${CMAKE_CURRENT_BINARY_DIR}/Textures_MediaStream.xbt
+    COMMAND ${TEXTUREPACKER_EXE} -input ${root}/addons/skin.mediastream/Media -output ${CMAKE_CURRENT_BINARY_DIR}/Textures_MediaStream.xbt
     MAIN_DEPENDENCY ${MEDIA_IMAGES_MEDIASTREAM}
     DEPENDS TexturePacker
     WORKING_DIRECTORY ${WORKDIR}
@@ -127,7 +133,7 @@
   if(ENABLE_NEW_SKIN)
     add_custom_command(
       OUTPUT Textures_Plex.xbt
-      COMMAND ${TEXTURE_PACKER_BIN} -input ${root}/addons/skin.plex/Media -output Textures_Plex.xbt
+      COMMAND ${TEXTUREPACKER_EXE} -input ${root}/addons/skin.plex/Media -output ${CMAKE_CURRENT_BINARY_DIR}/Textures_Plex.xbt
       MAIN_DEPENDENCY ${MEDIA_IMAGES_PLEX_SKIN}
       DEPENDS TexturePacker
     )
