--- plex/CMakeModules/PlatformConfigPOSIX.cmake	2013-12-13 09:47:18.000000000 +0100
+++ plex/CMakeModules/PlatformConfigPOSIX.cmake.new	2013-12-14 22:32:26.117329000 +0100
@@ -31,6 +31,12 @@
 set(HAS_SDL_JOYSTICK 1)
 set(HAS_LIBRTMP 1)
 
+if(NOT USE_INTERNAL_FFMPEG)
+  add_definitions(-DUSE_EXTERNAL_FFMPEG)
+else()
+  add_definitions(-DUSE_INTERNAL_FFMPEG)
+endif()
+
 add_definitions(
   -DTARGET_POSIX
   -D_REENTRANT
