--- plex/FileSystem/PlexDirectoryTypeParserPicture.cpp	2013-11-21 15:14:32.000000000 +0100
+++ plex/FileSystem/PlexDirectoryTypeParserPicture.cpp.new	2013-11-24 23:44:29.444208000 +0100
@@ -8,7 +8,7 @@
 
 #include "PlexDirectoryTypeParserPicture.h"
 #include "PlexAttributeParser.h"
-#include "Pictures/PictureInfoTag.h"
+#include "pictures/PictureInfoTag.h"
 
 void CPlexDirectoryTypeParserPicture::Process(CFileItem &item, CFileItem &mediaContainer, TiXmlElement *itemElement)
 {
