--- a/src/ui/webview.qml	2015-03-20 13:38:23.461304969 -0700
+++ b/src/ui/webview.qml	2015-03-20 13:42:07.625404038 -0700
@@ -60,15 +60,6 @@
       }
     }
 
-    MpvVideo
-    {
-      id: video
-      objectName: "video"
-      // It's not a real item. Its renderer draws onto the view's background.
-      width: 0
-      height: 0
-    }
-
     Timer {
       id: splashTimer
       interval: 1000
