diff --git a/src/ui/MainWindow.cpp b/src/ui/MainWindow.cpp
index e4292aa..053ce9e 100644
--- a/src/ui/MainWindow.cpp
+++ b/src/ui/MainWindow.cpp
@@ -81,14 +81,14 @@ void MainWindow::showEvent(QShowEvent *event)
   // This has to call after the mainwindow is inited since
   // the videoplayer init calls the mainwindow singelton.
   //
-  VideoPlayer* player = static_cast<VideoPlayer*>(ComponentManager::get()->getComponent("player"));
-  if (player)
-  {
-    player->setVideoSurface(m_videoSurface);
-    connect(player, &VideoPlayer::windowVisible, m_videoSurface, &VideoPlayerSurface::setOSDVisible);
-    connect(m_videoSurface, &VideoPlayerSurface::OSDFrameReady, player, &VideoPlayer::updateOSDOverlay);
-    connect(m_videoSurface, &VideoPlayerSurface::cursorVisibility, player, &VideoPlayer::setCursorVisible);
-  }
+//  VideoPlayer* player = static_cast<VideoPlayer*>(ComponentManager::get()->getComponent("player"));
+//  if (player)
+//  {
+//    player->setVideoSurface(m_videoSurface);
+//    connect(player, &VideoPlayer::windowVisible, m_videoSurface, &VideoPlayerSurface::setOSDVisible);
+//    connect(m_videoSurface, &VideoPlayerSurface::OSDFrameReady, player, &VideoPlayer::updateOSDOverlay);
+//    connect(m_videoSurface, &VideoPlayerSurface::cursorVisibility, player, &VideoPlayer::setCursorVisible);
+//  }
   
  return QMainWindow::showEvent(event);
 }
