--- a/Makefile	2015-02-04 13:49:55.000000000 -0800
+++ b/Makefile	2015-02-07 23:56:52.093893935 -0800
@@ -4,8 +4,6 @@
 
 LDFLAGS+=-L./ -Lffmpeg_compiled/usr/local/lib/ -lc -lWFC -lGLESv2 -lEGL -lbcm_host -lopenmaxil -lfreetype -lz
 
-INCLUDES+=-I./ -Ilinux -Iffmpeg_compiled/usr/local/include/ -I /usr/include/dbus-1.0 -I /usr/lib/arm-linux-gnueabihf/dbus-1.0/include
-
 DIST ?= omxplayer-dist
 
 SRC=linux/XMemUtils.cpp \
