--- a/CMakeLists.txt	2013-11-23 21:22:42.000000000 +0100
+++ b/CMakeLists.txt.new	2013-11-25 23:18:58.132208000 +0100
@@ -123,6 +123,39 @@
   ${root}/lib/libUPnP/Neptune/Source/System/Posix
 )
 
+# Add Python
+
+if(ENABLE_PYTHON)
+
+    if (TARGET_RPI)
+        install(DIRECTORY ${EXTERNAL_PYTHON_HOME}/lib/python2.7
+            DESTINATION ${LIBPATH}/lib/ COMPONENT RUNTIME 
+            PATTERN python2.7/test EXCLUDE
+            REGEX .*\\.pyc$ EXCLUDE
+            REGEX .*\\.pyo$ EXCLUDE
+        )
+        install(FILES
+        ${EXTERNAL_PYTHON_HOME}/include/python2.7/pyconfig.h DESTINATION
+        ${LIBPATH}/include/python2.7 COMPONENT RUNTIME)
+
+    elseif(TARGET_OSX)
+      install(DIRECTORY ${dependdir}/lib/python2.6 DESTINATION ${LIBPATH}/lib/ COMPONENT RUNTIME 
+        PATTERN python2.6/test EXCLUDE
+        REGEX .*\\.pyc$ EXCLUDE
+        REGEX .*\\.pyo$ EXCLUDE
+      )
+      install(FILES ${dependdir}/include/python2.6/pyconfig.h DESTINATION ${LIBPATH}/include/python2.6 COMPONENT RUNTIME)
+    elseif(TARGET_LINUX)
+      install(DIRECTORY ${EXTERNAL_PYTHON_HOME}/lib/python2.7 DESTINATION ${LIBPATH}/lib/ COMPONENT RUNTIME 
+        PATTERN python2.6/test EXCLUDE
+        REGEX .*\\.pyc$ EXCLUDE
+        REGEX .*\\.pyo$ EXCLUDE
+      )
+	install(FILES ${EXTERNAL_PYTHON_HOME}/include/python2.7/pyconfig.h DESTINATION ${LIBPATH}/include/python2.7 COMPONENT RUNTIME)
+    endif()
+
+endif()
+
 add_subdirectory(xbmc)
 if(TARGET_OSX)
   add_subdirectory(tools/EventClients/Clients/OSXRemote)
