--- a/xbmc/cdrip/EncoderFFmpeg.h	2015-03-12 20:52:41.054096999 +0100
+++ b/xbmc/cdrip/EncoderFFmpeg.h	2015-03-12 20:48:25.150096999 +0100
@@ -45,7 +45,15 @@
   AVCodecContext   *m_CodecCtx;
   AVStream         *m_Stream;
   AVPacket          m_Pkt;
-  unsigned char     m_BCBuffer[AVCODEC_MAX_AUDIO_FRAME_SIZE];
+  AVSampleFormat    m_InFormat;
+  AVSampleFormat    m_OutFormat;
+  /* From libavformat/avio.h:
+   * The buffer size is very important for performance.
+   * For protocols with fixed blocksize it should be set to this
+   * blocksize.
+   * For others a typical size is a cache page, e.g. 4kb.
+   */
+  unsigned char     m_BCBuffer[4096];
   static int        MuxerReadPacket(void *opaque, uint8_t *buf, int buf_size);
   void              SetTag(const CStdString tag, const CStdString value);
 
