--- a/qtwebkit/Source/WTF/wtf/Platform.h	2015-02-28 00:40:23.555370004 -0800
+++ b/qtwebkit/Source/WTF/wtf/Platform.h	2015-02-28 00:40:10.636059378 -0800
@@ -31,6 +31,7 @@
 /* Include compiler specific macros */
 #include <wtf/Compiler.h>
 
+#define ENABLE_LLINT 0
 
 /* ==== PLATFORM handles OS, operating environment, graphics API, and
    CPU. This macro will be phased out in favor of platform adaptation
