--- a/ext/openssl/extconf.rb	2013-10-28 07:32:24.000000000 +0100
+++ b/ext/openssl/extconf.rb.new	2015-03-12 10:37:09.298096999 +0100
@@ -100,9 +100,6 @@
 have_func("SSL_SESSION_get_id")
 have_func("SSL_SESSION_cmp")
 have_func("OPENSSL_cleanse")
-have_func("SSLv2_method")
-have_func("SSLv2_server_method")
-have_func("SSLv2_client_method")
 have_func("TLSv1_1_method")
 have_func("TLSv1_1_server_method")
 have_func("TLSv1_1_client_method")
