--- a/qtwebengine/src/core/config/embedded_linux.pri	2015-09-04 14:41:04.000000000 +0200
+++ b/qtwebengine/src/core/config/embedded_linux.pri	2015-09-23 20:23:44.800000000 +0200
@@ -46,7 +46,7 @@
     use_ozone=1 \
     use_pango=0 \
     use_system_fontconfig=1 \
-    icu_use_data_file_flag=0 \
+    icu_use_data_file_flag=1 \
     use_x11=0 \
     v8_use_snapshot=false \
     want_separate_host_toolset=1 \
